### PR TITLE
Allow external payment addresses in Trezor CIP-36 registrations

### PR DESCRIPTION
**Pull request by davidmisiak**

### CODE REVIEW

File: src/crypto-providers/trezorCryptoProvider.ts [hunk 1]
```diff
@@ -211,6 +219,27 @@ const TrezorCryptoProvider: () => Promise<CryptoProvider> = async () => {
       }
     })
 
+  const prepareDestination = (
+    address: Buffer,
+    changeAddressParams: _AddressParameters | null,
```
Comment (janmazak) [inline src/crypto-providers/trezorCryptoProvider.ts:224]: Perhaps just `addressParams` without "change"?

File: src/crypto-providers/trezorCryptoProvider.ts [hunk 2]
```diff
@@ -741,53 +743,39 @@ const TrezorCryptoProvider: () => Promise<CryptoProvider> = async () => {
...
-          stakingPath: addressParameters.stakePath as BIP32Path,
-        }
-      }
-      throw Error(Errors.InvalidCIP36RegistrationAddressType)
+    const destination: {
+      address?: string
+      addressParameters?: TrezorTypes.CardanoAddressParameters
+    } = paymentDestination
+    if (Number(votingPurpose) > Number.MAX_SAFE_INTEGER) {
+      throw Error(Errors.InvalidCVoteVotingPurpose)
```
Comment (janmazak) [inline src/crypto-providers/trezorCryptoProvider.ts:769]: Are we dropping support for reward addresses? I guess it's OK for future use, though Ledger only gives a warning in that case.

**Review by janmazak — APPROVED** [review]

File: src/crypto-providers/trezorCryptoProvider.ts (hunk 1 above)
Comment (davidmisiak) [reply src/crypto-providers/trezorCryptoProvider.ts:224]: Below in the function, we check that the `paymentPath` is present, so I think that `changeAddressParams` is appropriate.

File: src/crypto-providers/trezorCryptoProvider.ts (hunk 2 above)
Comment (davidmisiak) [reply src/crypto-providers/trezorCryptoProvider.ts:769]: I don't think so - e.g. most test cases still use the reward address. The code should be equivalent to ledgerCryptoProvider.

Am I missing something?

**Pull request merged**